rest/route/project_validate.go

```diff
@@ -0,0 +1,94 @@
+package route
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/validator"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/utility"
+	"github.com/pkg/errors"
+)
+
+////////////////////////////////////////////////////////////////////////
+//
+// POST /rest/v2/validate
+
+type validateProjectHandler struct {
+	input validator.ValidationInput
+}
+
+func makeValidateProject() gimlet.RouteHandler {
+	return &validateProjectHandler{}
+}
+
+// Factory creates an instance of the handler.
+//
+//	@Summary		Validate a project
+//	@Description	Validate a project configuration file.
+//	@Tags			projects
+//	@Router			/validate [post]
+//	@Security		Api-User || Api-Key
+//	@Param			{object}	body		validator.ValidationInput	true	"parameters"
+//	@Success		200			{object}	validator.ValidationErrors
+func (v *validateProjectHandler) Factory() gimlet.RouteHandler {
+	return &validateProjectHandler{}
+}
+
+func (v *validateProjectHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := utility.NewRequestReader(r)
+	defer body.Close()
+
+	bytes, err := io.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Error reading request body")
+	}
+
+	if !json.Valid(bytes) {
+		return errors.New("Invalid JSON format detected; potential incomplete or corrupted data. This may be an indication that evergreen is under high load.")
+	}
+
+	input := validator.ValidationInput{}
+
+	if err := json.Unmarshal(bytes, &input); err != nil {
+		return errors.Wrap(err, "Unmarshaling request body")
+	}
+
+	v.input = input
+	return nil
+}
+
+func (v *validateProjectHandler) Run(ctx context.Context) gimlet.Responder {
+
+	project := &model.Project{}
+	var projectConfig *model.ProjectConfig
+	var err error
+
+	opts := &model.GetProjectOpts{
+		ReadFileFrom: model.ReadFromLocal,
+	}
+	validationErr := validator.ValidationError{}
+	if _, err = model.LoadProjectInto(ctx, v.input.ProjectYaml, opts, "", project); err != nil {
```

Comment:
can we add the project id into LoadProjectInto 
(not setting project id could break single task distro validation)